### ISSUE TITLE
ChartCanvasLinePlot should support "holes" in the data

### ISSUE THREAD

**Issue by samreid** [phetsims/bamboo#17]
LinePlot supports missing data like so:

```js
      // NaN or Infinite components "pen up" and stop drawing
      if ( this.dataSet[ i ].isFinite() ) {
```

It seems ChartCanvasLinePlot should support it too, so it is easy to switch between plot types.

Discovered in #16

**Comment by pixelzoom** [comment]
This feature also needs to be documented in LinePlot.  I didn't know this feature existed, and would only have discovered it by reading the implementation of `updateLinePlot`.

**Comment by samreid** [comment]
I updated CanvasLinePlot so it skips non-finite data the same way LinePlot does, and I added a note of documentation to each.  @pixelzoom can you please review?

**Comment by pixelzoom** [comment]
This still didn't seem clear to me, so I took a stab at more clarification and examples in the above commit.  

In doing this, I also discovered another issue.  A `dataSet` is consistently described as a `{Vector2[]}`:

```js
 * LinePlot renders a dataset of {Vector2[]} by connecting the points with line segments.
...
@param {Vector2[]} dataSet
```

But these non-finite values that can be inserted into a data set are _NOT_ Vector2.   So should the type expression throughout be changed to `{Array.<Vector2|number>}`, with the constraint the a `{number}` must be non-finite?   This also makes me think that using non-finite numbers is the wrong approach. Why not use `null`?  It would be much easier to describe, the type expression would be `{Array.<Vector2|null>}`, and an example would look like:

`[ (0,0), (0,1), null, (0,2), (0,3) ]`

**Comment by samreid** [comment]
The current implementation only supports `Vector2[]` and uses `Vector2.isFinite()` to determine if a value will be rendered or not.  This is a natural fit for time series data, where one dimension is known and one may be undefined, such as `new Vector2(7.5, NaN)`.  This strategy is used in both Wave Interference and Circuit Construction Kit.  For bamboo it doesn't seem important to know that a measurement was NaN at a certain time, maybe other situations would leverage that (or maybe not).

On the other hand, an array of finite Vector2 | null also seems reasonable.  Would we want to collapse multiple `null` values?  For the purposes of this chart, `[ (0,0), (0,1), null, (0,2), (0,3) ]` would render the same as `[ (0,0), (0,1), null, null, (0,2), (0,3) ]`.

**Comment by pixelzoom** [comment]
Ah, I see -- `this.dataSet[ i ].isFinite()` still involves a Vector2.  Thanks for clarifying.  I've fixed the doc in the above commit.

It seems odd (and as I've unintentionally demonstrated, confusing) to have to create a non-finite Vector2 instance -- and either know or investigate how Vector2 `isFinite` is implemented.  Using `null` seem clearer, easier to explain, and creates fewer objects.  If we switched to using `null` to represent a gap, I see no reason to collapse consecutive `null` values. 

Also... The doc for Vector2 `isFinite` is a tad confusing:

```js
* Whether all of the components are numbers (not NaN) that are not infinity or -infinity.
```

Better would be:

```js
 * Returns false if either component is NaN, infinity, or -infinity. Otherwise returns true.
```

**Comment by samreid** [comment]
I revised the documentation as recommended, and converted to use `null` instead of non-finite `Vector2` for the data.  I found that CCK was using these renderers from bamboo, but Wave Interference is not, so Wave Interference didn't need to be updated.  I was concerned there would be a problem in CCK because it used the "time" value of a non-finite Vector2 to know when to remove a data point from the data set, but I tested removing null points and didn't see any problems, and it seems like it will be OK.

Ready for review.
